runtime/runtime-params-estimator/estimator-warehouse/src/check.rs

```diff
@@ -92,34 +68,227 @@ pub(crate) fn check(db: &Db, config: &CheckConfig) -> anyhow::Result<()> {
+}
+
 fn estimation_changes(
     db: &Db,
     estimation_names: &[String],
-    commit_a: &str,
-    commit_b: &str,
+    commit_before: &str,
+    commit_after: &str,
     tolerance: f64,
     metric: Metric,
 ) -> anyhow::Result<Vec<Notice>> {
     let mut warnings = Vec::new();
     for name in estimation_names {
-        let a = &EstimationRow::get(db, name, commit_a, metric)?[0];
-        let b = &EstimationRow::get(db, name, commit_b, metric)?[0];
-        let rel_change = if a.gas == 0.0 { 100.0 } else { (a.gas - b.gas).abs() / a.gas };
+        let b = &EstimationRow::get(db, name, commit_before, metric)?[0];
+        let a = &EstimationRow::get(db, name, commit_after, metric)?[0];
+        let rel_change = if b.gas == 0.0 { 1.0 } else { (b.gas - a.gas).abs() / b.gas };
         if rel_change > tolerance {
             warnings.push(Notice::RelativeChange(RelativeChange {
                 estimation: name.clone(),
-                before: a.gas,
-                after: b.gas,
+                before: b.gas,
+                after: a.gas,
             }))
         }
     }
 
     Ok(warnings)
 }
+
+#[cfg(test)]
+mod test {
+    use super::{inner_check, CheckConfig, Notice, RelativeChange};
+    use crate::db::{Db, EstimationRow};
+    use crate::Metric;
+
+    #[track_caller]
+    fn check_estimation_changes(
+        db: &Db,
+        estimation_names: impl Iterator<Item = &'static str>,
+        commit_before: &str,
+        commit_after: &str,
+        tolerance: f64,
+        metric: Metric,
+        expected: Vec<Notice>,
+    ) {
+        let output = super::estimation_changes(
+            &db,
+            &estimation_names.map(|s| s.to_owned()).collect::<Vec<_>>(),
+            commit_before,
+            commit_after,
+            tolerance,
+            metric,
+        )
+        .unwrap();
+        assert_eq!(expected, output);
+    }
+
+    #[test]
+    fn test_estimation_changes() {
+        let db = Db::test();
+        // Data with large difference between commits
+        EstimationRow::insert_test_data_set(&db, "LogBase", 1e9, 1e9, Metric::ICount, 3);
+        // Data with virtually no difference between commits
+        EstimationRow::insert_test_data_set(&db, "LogByte", 1e9, 1.0, Metric::ICount, 3);
+        EstimationRow::insert_test_data_set(&db, "LogBase", 2e9, 0.0, Metric::Time, 3);
+        EstimationRow::insert_test_data_set(&db, "LogByte", 2e9, 0.0, Metric::Time, 3);
+
+        let expected_icount_notice = Notice::RelativeChange(RelativeChange {
+            estimation: "LogBase".to_owned(),
+            before: 2e9,
+            after: 3e9,
+        });
+
+        // The difference is exactly 50%, therefore 49.9% is breached.
+        check_estimation_changes(
+            &db,
+            ["LogBase", "LogByte"].into_iter(),
+            "0001beef",
+            "0002beef",
+            0.499,
+            Metric::ICount,
+            vec![expected_icount_notice],
```

Comment:
For `expected`, I'd use string:

* less types to change
* more legible
* actually tests what we care about -- the sting in the zulip message.